maths/simpson_rule.py

```diff
@@ -1,17 +1,59 @@
 """
 Numerical integration or quadrature for a smooth function f with known values at x_i
 
-This method is the classical approach of suming 'Equally Spaced Abscissas'
+This method is the classical approach of summing 'Equally Spaced Abscissas'
 
 method 2:
 "Simpson Rule"
 
 """
 
 
-def method_2(boundary, steps):
+def method_2(boundary: list[int], steps: int) -> float:
     # "Simpson Rule"
     # int(f) = delta_x/2 * (b-a)/3*(f1 + 4f2 + 2f_3 + ... + fn)
+    """
+    Calculate the definite integral of a function using Simpson's Rule.
+
+     :param boundary: A list containing the lower and upper bounds of integration.
+     :param steps: The number of steps or resolution for the integration.
+     :return: The approximate integral value.
+
+        >>> round(method_2([0, 2, 4], 10), 10)
+        2.6666666667
+
+        >>> round(method_2([2, 0], 10), 10)
+        -0.2666666667
+
+        >>> round(method_2([-2, -1], 10), 10)
+        2.172
+
+        # Test with a linear function f(x) = x
+        >>> round(method_2([0, 1], 10), 10)
+        0.3333333333
+
+        # Test with a constant function f(x) = 5
+        >>> round(method_2([0, 2], 10), 10)
+        2.6666666667
+
+        # Test with a quadratic function f(x) = x^2
+        >>> round(method_2([0, 2], 100), 10)
+        2.5621226667
+
+        # Test with a cubic function f(x) = x^3
+        >>> round(method_2([0, 1], 1000), 10)
+        0.3320026653
+
+        # Test with invalid input
```

Comment:
It is Pythonic to raise a ValueError when we receive invalid input.  This repo contains tons of examples of how to do this.  https://github.com/search?q=repo%3ATheAlgorithms%2FPython%20ValueError&type=code